client/client.go

```diff
@@ -137,47 +193,33 @@ func decodeHeader(header []byte) (code []byte, pt uint32, length int) {
 }
 
 func (client *Client) reconnect(err error) error {
-	if client.conn != nil {
-		return nil
-	}
 
-	// TODO I doubt this error handling is right because it looks
-	// really complicated.
+	// not actioning on error if it's deemed Temporary
+	// we might want to take note of timestamp and eventually recycle this connection
+	// if it persists too long (even though classified as Temporary here)
 	if opErr, ok := err.(*net.OpError); ok {
-		if opErr.Timeout() {
-			client.err(err)
-		}
 		if opErr.Temporary() {
 			return nil
 		}
-
-		return err
-	}
-
-	if err != nil {
-		client.err(err)
 	}
 
 	// If it is unexpected error and the connection wasn't
 	// closed by Gearmand, the client should close the conection
 	// and reconnect to job server.
-	client.Close()
```

Comment:
If you are removing this line, please remove the comment above.

Reply:
fixed